### PR TITLE
Meta: Replace IPC linter with something that works

### PR DESCRIPTION
**Pull request by BenWiederhake**
The old `Meta/lint-ipc-ids.sh` script was silently broken in 62af6cd4f9637b8937e59832f5af00f4556c496b. This PR mainly removes the script, and replaces it by a tool that runs similarly to the Markdown checker. Specifically:
- Remove hard-coded magic numbers and the syntax for it (this also simplifies the syntax so that it is easier to parse to name)
- Implement the linter itself
- Run it during CI and pre-commit
- Remove the old script

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
English nit-pick: You say "replace foo _with_ x" not "replace foo _by_ x". (You would use "by" when it's followed by a -ing verb (gerund?), like "replace foo by generating bars", but that's not common.)

Not important but thought I'd say anyway. :^)

File: Meta/Lagom/Tools/CMakeLists.txt [hunk 1]
```diff
@@ -11,5 +11,6 @@ function(lagom_tool tool)
     )
 endfunction()
 
-add_subdirectory(ConfigureComponents)
 add_subdirectory(CodeGenerators)
+add_subdirectory(ConfigureComponents)
+add_subdirectory(IpcMagicLinter)
```
[the remark this answers is not in the document]
Comment (BenWiederhake) [reply Meta/Lagom/Tools/CMakeLists.txt:16]: Currently, the naming is all over the place. As the most egregious example, we have both `JSONValue.h` and `JsonValue.h` in the repo. I was hoping to pull the tow in this tug of war a bit in the `JsonValue.h` direction :D

Yes, the names are JSON, HTML, and IPC. And yet I suggest to name the classes JsonValue, HtmlElement, and IpcMagicLinter.

Yes, no, maybe? :)

File: Meta/Lagom/Tools/IpcMagicLinter/main.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        warnln("Usage: {} path/to/some.ipc path/to/other.ipc [more ipc files ...]", argv[0]);
+        return EXIT_ERROR;
+    }
+
+    // Read files, compute their hashes, ignore collisions for now.
+    HashMap<u32, Vector<String>> inverse_hashes;
+    bool had_errors = false;
+    for (int file_index = 1; file_index < argc; ++file_index) {
+        String filename(argv[file_index]);
+        auto file_or_error = Core::File::open(argv[file_index], Core::OpenMode::ReadOnly);
```
[the remark this answers is not in the document]
Comment (BenWiederhake) [reply Meta/Lagom/Tools/IpcMagicLinter/main.cpp:28]: Doink, you're right! :D

**Comment by BenWiederhake** [comment]
Changes since last push:
- Use a String where the String was already available (thanks Ali!)
- Add new commit that changes how the endpoint magic is treated in gethostbyname and gethostbyaddr
- Rebase for convenience (and to resolve conflict with e.g. c2d44209a8f508ccb243410fc0e591976820b287)
- Make extra sure it really doesn't break:

Bildschirmfoto_2021-11-03_23-04-10

File: Meta/Lagom/Tools/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (alimpfard) [reply Meta/Lagom/Tools/CMakeLists.txt:16]: I'll take that, seems fine - even if "Ipc" is weird and sounds like a prefix of a drug's name.

**Comment by BenWiederhake** [comment]
Changes since last push:
- Renamed from IpcMagicLinter to IPCMagicLinter (I object to this naming scheme, but whatever)
- Fixed the copyright declaration in `Meta/Lagom/Tools/IPCMagicLinter/main.cpp` (d'oh!)
- Rebased

**Pull request merged**